sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/util/RLimit.java

```diff
@@ -0,0 +1,87 @@
+package com.mesosphere.sdk.specification.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.mesos.Protos;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Representation of an individual rlimit, consisting of a name and optional soft/hard limits.
+ *
+ * A valid instance of this class has either both limits set or neither, with the further constraint that the soft limit
+ * must be less than or equal to the hard limit.
+ */
+public class RLimit {
+    private static final Set<String> RLIMITS = new HashSet<>();
+
+    static {
+        for (Protos.RLimitInfo.RLimit.Type rLimitType : Protos.RLimitInfo.RLimit.Type.values()) {
+            RLIMITS.add(rLimitType.toString());
```

Comment:
If you wanted to support e.g. `AS` instead of requiring `RLMT_AS` for convenience, the map (once moved from `DefaultOfferRequirementProvider`) could strip out the `RLMT_` on names when present:

```
for (Protos.RLimitInfo.RLimit.Type rLimitType : Protos.RLimitInfo.RLimit.Type.values()) {
    String typeName = rLimitType.toString();
    if (typeName.startsWith("RLMT_")) {
        typeName = typeName.substring(5);
    }
    RLIMIT_TYPE_MAP.put(typeName, rLimitType);
}
```

Or (maybe better) the mesos `RLMT_` prefix could be replaced with `RLIMIT_` so that names align with docs: https://linux.die.net/man/2/setrlimit